lib/rubocop/cop/variable_force/variable_table.rb

```diff
@@ -113,14 +113,14 @@ def variable_exist?(name)
         def accessible_variables
           scope_stack.reverse_each.with_object([]) do |scope, variables|
             variables.concat(scope.variables.values)
-            break variables unless scope.node.block_type?
+            break variables unless scope.node.block_type? || scope.node.numblock_type?
```

Comment:
As mentioned in the commit message, I tentatively added this to match the other `block_type?` call sites in this file:

- #11485 added it in `find_variable` above, to resolve a false positive (#11485)
- I am adding it in `mark_variable_as_captured_by_block_if_so` below, to resolve a false positive

I don't know what the false positive is here, but I'm assuming there is one. If someone with more context knows what it might be, I'm can add a corresponding test and update the commit message and changelog.